Source/WebInspectorUI/UserInterface/Base/Main.js

```diff
@@ -1821,22 +1822,18 @@ WI._saveCookieForOpenTabs = function()
     }
 };
 
-WI._windowFocused = function(event)
+WI._updateWindowInactiveState = function(event)
 {
-    if (event.target.document.nodeType !== Node.DOCUMENT_NODE)
-        return;
-
     // FIXME: We should use the :window-inactive pseudo class once https://webkit.org/b/38927 is fixed.
-    document.body.classList.remove(WI.dockConfiguration === WI.DockConfiguration.Undocked ? "window-inactive" : "window-docked-inactive");
-};
 
-WI._windowBlurred = function(event)
-{
-    if (event.target.document.nodeType !== Node.DOCUMENT_NODE)
-        return;
+    if (document.activeElement?.tagName === "IFRAME") {
+        // An active iframe means an extension tab is active and we can't tell when the window blurs due to cross-origin restrictions.
+        // In this case we need to keep checking to know if the window looses focus since there is no event we can use.
+        window.requestAnimationFrame(WI._updateWindowInactiveState);
```

Comment:
oof, it kinda sucks to have to re-check every single rAF :/

can we maybe do some sort of rate limiting (i.e. once per second) or something?

Reply:
I wanted to make sure it switched at the same time, since the delay was very noticeable when I tried a rate limited approach.

Reply:
not even like 100ms/50ms/etc.?  can we at least get some sort of measurement that this doesn't negatively impact things?

what about adding some `InspectorFrontendHost` method to allow us to get notified when the `<iframe>` changes?

Reply:
I rechecked with a release build, and using a 250ms timeout was fine visually, so I've updated the PR with that.